src/lib/ext-popup.js

```diff
@@ -0,0 +1,105 @@
+import {
+  getMonitorStatus,
+  startMonitor,
+  stopMonitor,
+  openActivityLogPage,
+} from '../lib/ext-listen.js';
+
+export default class Popup {
+  constructor() {
+    this.startMonitorAllBtn = document.getElementById('startMonitorBtn');
+    this.stopMonitorAllBtn = document.getElementById('stopMonitorBtn');
+    this.openActivityLogBtn = document.getElementById('actLogPage');
+    this.monitorStatusText = document.getElementById('monitorStatus');
+    this.errorMsgText = document.getElementById('errorText');
+  }
+
+  renderErrorMsg(message) {
+    this.errorMsgText.textContent = message;
+    this.errorMsgText.style.display = 'block';
+  }
+
+  renderMonitorStartedUI() {
+    this.startMonitorAllBtn.setAttribute('disabled', 'disabled');
+    this.stopMonitorAllBtn.removeAttribute('disabled');
+
+    this.monitorStatusText.textContent = 'Extensions are being monitored';
+    this.monitorStatusText.classList.add('success');
+    this.monitorStatusText.classList.remove('failure');
+  }
+
+  renderMonitorStoppedUI() {
+    this.startMonitorAllBtn.removeAttribute('disabled');
+    this.stopMonitorAllBtn.setAttribute('disabled', 'disabled');
+
+    this.monitorStatusText.textContent = 'No extensions are being monitored';
+    this.monitorStatusText.classList.add('failure');
+    this.monitorStatusText.classList.remove('success');
+  }
+
+  async render() {
+    const isMonitorStarted = await getMonitorStatus();
+    isMonitorStarted
+      ? this.renderMonitorStartedUI()
+      : this.renderMonitorStoppedUI();
+  }
+
+  handleViewActivityLog() {
+    openActivityLogPage();
+    window.close();
+  }
+
+  async handleEvent(event) {
+    if (event.type === 'click') {
+      switch (event.target) {
+        case this.startMonitorAllBtn:
+          await this.handleStartMonitor();
+          break;
+        case this.stopMonitorAllBtn:
+          await this.handleStopMonitor();
+          break;
+        case this.openActivityLogBtn:
+          this.handleViewActivityLog();
+          break;
+        default:
+          throw new Error('wrong event target found ' + event.target);
+      }
+    } else {
+      throw new Error('wrong event type found ' + event.type);
+    }
+  }
+
+  async handleStartMonitor() {
+    try {
+      const monitorMsg = await startMonitor();
+      if (monitorMsg) {
+        this.render();
+      } else {
+        throw new Error("Monitoring couldn't be started");
+      }
```

Comment:
Nit, It doesn't seem that startMonitor can actually ever return false (it rejects if it can't start the monitor because it is already active), and so this may be likely simplified to just do:
```js
try {
  await startMonitor();
  this.render();
} catch (error) {
  this.renderErrorMsg(error.message);
}
```

same in handleStopMonitor.